### PR TITLE
Start Converting Com Vtbls to Cswin32

### PR DESCRIPTION
**Pull request by lonitra**
- Convert `IStreamVtbl`,`IFileDialogEventsVtbl`,`IEnumStringVtbl`,`IEnumFORMATETCVtble`,
`IDropSourceVtbl`,`IDropSourceNotifyVtbl`
- Add implicit operators for `Exception` > `HRESULT` and  `ComType.FORMATETC` > `FORMATETC`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7906)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.IDropSourceVtbl.cs [hunk 1]
```diff
@@ -14,43 +15,41 @@ internal static class IDropSourceVtbl
...
             public static IntPtr Create(IntPtr fpQueryInterface, IntPtr fpAddRef, IntPtr fpRelease)
             {
-                IntPtr* vtblRaw = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IDropSourceVtbl), IntPtr.Size * 5);
-                vtblRaw[0] = fpQueryInterface;
-                vtblRaw[1] = fpAddRef;
-                vtblRaw[2] = fpRelease;
-                vtblRaw[3] = (IntPtr)(delegate* unmanaged<IntPtr, BOOL, User32.MK, HRESULT>)&QueryContinueDrag;
-                vtblRaw[4] = (IntPtr)(delegate* unmanaged<IntPtr, Ole32.DROPEFFECT, HRESULT>)&GiveFeedback;
+                IDropSource.Vtbl* vtblRaw = (IDropSource.Vtbl*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IDropSourceVtbl), sizeof(IDropSource.Vtbl));
+                vtblRaw->QueryInterface_1 = (delegate* unmanaged[Stdcall]<IDropSource*, Guid*, void**, HRESULT>)fpQueryInterface;
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.IDropSourceVtbl.cs:19]: I think we can improve this experience. `Create()` should probably not take these three pointers and should return `IUnknown*` and we can hook them in the calling code with a helper method (`AddIUnknownImpl(IUnknown*)`). We can look at this together as a follow-up.

**Review by JeremyKuhne — APPROVED** [review]
Looks good!

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/Foundation/HRESULT.cs [hunk 2]
```diff
@@ -2,11 +2,19 @@
...
+using System.Diagnostics;
+
 namespace Windows.Win32.Foundation
 {
     internal readonly partial struct HRESULT
     {
         public static HRESULT HRESULT_FROM_WIN32(WIN32_ERROR error)
             => new(((int)error & 0x0000FFFF) | unchecked((int)0x80070000));
+
+        public static implicit operator HRESULT(Exception ex)
```
Comment (kant2002) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/Foundation/HRESULT.cs:14]: This one lead to very puzzling view of the code.

**Pull request merged**